### PR TITLE
fix: ensure to correctly encode binary data as base64

### PR DESCRIPTION
**Pull request by moolen**
Fixes: #2673
Fixes: #2375

### CODE REVIEW

File: pkg/provider/kubernetes/client_test.go [hunk 1]
```diff
@@ -191,6 +194,69 @@ func TestGetSecret(t *testing.T) {
...
+						},
+					},
+				},
+				Namespace: "default",
+			},
+			ref: esv1beta1.ExternalSecretDataRemoteRef{
+				Key:      "mysec",
+				Property: "bindata",
+			},
+			want: []byte(base64.StdEncoding.EncodeToString(binaryTestData)),
```
Comment (moolen) [inline pkg/provider/kubernetes/client_test.go:258]: This previously was encoded as string `"\uffed\uffed..."` which was absolutely unusable, see linked issue above.

**Comment by gusfcarvalho** [comment]
I wonder if this would break any secrets that actually have HTML pages 😄 from my understanding of the code, I think not.

**Comment by moolen** [comment]
From the docs:
```go
// String values encode as JSON strings coerced to valid UTF-8,
// replacing invalid bytes with the Unicode replacement rune.
// So that the JSON will be safe to embed inside HTML <script> tags,
// the string is encoded using HTMLEscape,
// which replaces "<", ">", "&", U+2028, and U+2029 are escaped
// to "\u003c","\u003e", "\u0026", "\u2028", and "\u2029".
// This replacement can be disabled when using an Encoder,
// by calling SetEscapeHTML(false).
```

So this replacement won't happen any more. There could be secrets that contain :point_up_2: characters. I'll add a :warning: info to the release notes about this (bugfix) breaking change.

**Pull request merged**